test/src/main/java/org/apache/accumulo/test/ScanServerIT.java

```diff
@@ -224,6 +234,110 @@ public void testBatchScannerTimeout() throws Exception {
+            "The scan server scanner should have seen all ingested and flushed entries");
+        // Throws an exception because of the tablets with the NEVER hosting goal
+        scanner.setConsistencyLevel(ConsistencyLevel.IMMEDIATE);
+        assertThrows(RuntimeException.class, () -> Iterables.size(scanner));
+      } // when the scanner is closed, all open sessions should be closed
+    }
+  }
+
+  @Test
+  public void testBatchScanWithTabletHostingMix() throws Exception {
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+      String tableName = getUniqueNames(1)[0];
+
+      SortedSet<Text> splits = new TreeSet<>();
+      splits.add(new Text("row_0000000001"));
+      splits.add(new Text("row_0000000002"));
+      splits.add(new Text("row_0000000003"));
+      splits.add(new Text("row_0000000004"));
+      splits.add(new Text("row_0000000005"));
+      splits.add(new Text("row_0000000006"));
+      splits.add(new Text("row_0000000007"));
+      splits.add(new Text("row_0000000008"));
+      splits.add(new Text("row_0000000009"));
+      NewTableConfiguration ntc = new NewTableConfiguration();
+      ntc.withSplits(splits);
+      ntc.withInitialHostingGoal(TabletHostingGoal.ALWAYS); // speed up ingest
+      final int ingestedEntryCount = createTableAndIngest(client, tableName, ntc, 10, 10, "colf");
+
+      String tableId = client.tableOperations().tableIdMap().get(tableName);
+
+      // row 1 -> 3 are always
+      client.tableOperations().setTabletHostingGoal(tableName,
+          new Range(null, true, "row_0000000003", true), TabletHostingGoal.ALWAYS);
+      // row 4 -> 7 are never
+      client.tableOperations().setTabletHostingGoal(tableName,
+          new Range("row_0000000004", true, "row_0000000007", true), TabletHostingGoal.NEVER);
+      // row 8 and 9 are ondemand
+      client.tableOperations().setTabletHostingGoal(tableName,
+          new Range("row_0000000008", true, null, true), TabletHostingGoal.ONDEMAND);
+
+      // Wait for the NEVER and ONDEMAND tablets to be unloaded
+      int hosted = getNumHostedTablets(client, tableId);
+      // Waiting for tablets to be unloaded due to inactivity
+      while (hosted != 3) {
+        Thread.sleep(1000);
+        hosted = getNumHostedTablets(client, tableId);
+      }
+
+      try (BatchScanner scanner = client.createBatchScanner(tableName, Authorizations.EMPTY)) {
+        scanner.setRanges(Collections.singleton(new Range()));
+        scanner.setConsistencyLevel(ConsistencyLevel.EVENTUAL);
+        assertEquals(ingestedEntryCount, Iterables.size(scanner),
+            "The scan server scanner should have seen all ingested and flushed entries");
+        // Throws an exception because of the tablets with the NEVER hosting goal
+        scanner.setConsistencyLevel(ConsistencyLevel.IMMEDIATE);
+        assertThrows(RuntimeException.class, () -> Iterables.size(scanner));
```

Comment:
Could also try scanning the ranges of the table that should work.

```suggestion
        assertThrows(RuntimeException.class, () -> Iterables.size(scanner));
        
        scanner.setRange(new Range(null,"row_0000000003"));
        assertEquals(???, Iterables.size(scanner));
        
         scanner.setRange(new Range("row_0000000008",null));
         assertEquals(???, Iterables.size(scanner));
```

Reply:
The above comment was meant for the scanner test, but I made it in the batch scanner test.  So the set range call is not correct.

Reply:
Added in cabcf0b